### PR TITLE
Improving keyboard input

### PR DESCRIPTION
**Pull request by Shom770**
Using ``threading`` with a while True loop will greatly slow down our code as it needs to request the GIL to do so every iteration.
With that in mind, I switched the keyboard input portion to ``pynput`` as ``blessed`` is quite slow with keyboard input too.
For those who run on Linux, please run this before reviewing so you can let me know if it doesn't work on Linux

### CODE REVIEW

File: src/resources/entities/character.py [hunk 1]
```diff
@@ -25,34 +23,44 @@ def __init__(self, current_level: Level, symbol: str = "$") -> None:
...
-                    on = False
-                    self.playing = False
-                self.ground_symbol = self.level.board[self.y][self.x]
+        """Uses listener that reads keyboard input from press"""
+        with Listener(on_press=self.press, on_release=self.release) as listener:  # set keys to be read immediately
+            listener.join()
 
     def draw(self) -> None:
         """Places player on map"""
+        print((self.y, self.x))
```
Comment (cons01e) [inline src/resources/entities/character.py]: Remove this print statement.

File: src/resources/entities/character.py [hunk 2]
```diff
@@ -25,34 +23,44 @@ def __init__(self, current_level: Level, symbol: str = "$") -> None:
...
+                return False
+            self.ground_symbol = self.level.board[self.y][self.x]
+        except AttributeError:
+            self.playing = False
+
+    def release(self, key: Key) -> bool:
+        """On key release"""
+        try:
+            if key.char not in ('w', 'a', 's', 'd', 'p'):
+                self.playing = False
```
Comment (cons01e) [inline src/resources/entities/character.py]: We should have a more specific exit condition. Adding new user buttons will require coming here to change this list which will become unmanageable and is also error prone. For now just pick a button as the exit condition and check if the user click it.

File: src/resources/entities/character.py [hunk 3]
```diff
@@ -25,34 +23,44 @@ def __init__(self, current_level: Level, symbol: str = "$") -> None:
...
+        except AttributeError:
+            self.playing = False
```
Comment (cons01e) [inline src/resources/entities/character.py]: Add a message indicating an error before kicking the player out of the game.

**Review by cons01e — CHANGES_REQUESTED** [review]
Good work! 
The controls still struggle when I click the buttons quickly but overall it seems to work.

File: src/resources/entities/character.py (hunk 2 above)
Comment (n0remac) [reply src/resources/entities/character.py]: Originally I had only mapped p to close the game. Any button will do

**Comment by Shom770** [comment]
> Good work!
> The controls still struggle when I click the buttons quickly but overall it seems to work.

Thanks! The changes you requested have been fixed, and the last one may be due to the slow refreshing of the layout (10 FPS)

**Review by n0remac — APPROVED** [review]

**Pull request merged**